### ISSUE TITLE
Intersector.intersectPolygons fails

### ISSUE THREAD

**Issue by kalayda** [libgdx/libgdx#7454]
Using `Intersector.intersectPolygons` to find intersection of square and rotated trapezoid gives wrong overlapping.

In code
```
val trapezoid = Polygon(FloatArray(8))
val square = Polygon(FloatArray(8))
val overlap = Polygon()
// Initialize trapezoid.vertices and square.vertices with xy cordinates in CLOCKWISE order.
Intersector.intersectPolygons(trapezoid, square, overlap)
```
When initializing `trapezoid` with `-2.8626027, -3.054608, 0.09345288, 14.654785, 18.01647, 4.9480224, 4.8018613, -7.2055407`  
and `square` with `-8.879811, -14.852028, -8.879811, 5.1479716, 11.120189, 5.1479716, 11.120189, -14.852028`,  
I got `overlap` of  
`-3.4702098, 5.220678, -3.470294, 5.2206793, 0.0, 0.0, 3.4595287, 5.220678, 3.4597168, 5.220703, 0.0, 0.0`

Here is grafical image of input objects
![image](https://github.com/user-attachments/assets/3c0d9114-580b-4468-8b38-2207e165930f)

There is blue trapezoid intersecting with light-gray square, and YELLOW `overlap` result.

What can be wrong?

**Comment by tommyettinger** [comment]
I did my best to reproduce your conditions, but did not get anything like what you showed:

![java_FXXqlCUdrH](https://github.com/user-attachments/assets/d57e377e-c52d-43e9-b9b4-5d0d8db754c6)

[Here's the code I used](https://github.com/yellowstonegames/SquidSquad/blob/ba291ef2cba3afbd6691ca84f6d0267fc1571c2a/squidworld/src/test/java/com/github/yellowstonegames/world/IntersectorTest.java); it uses some libraries that I already was using, because I just wanted to use what I knew and get this tested quickly.

The `overlap` Polygon is drawn in red here, and seems correct, so it definitely isn't doing what your code does. We probably would need a simple, self-contained example that demonstrates the exact problem, and you haven't yet provided that. It seems like there could be some unrelated code that is also writing to the overlap Polygon.
